qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py

```diff
@@ -15,13 +15,94 @@
 File System Model for QLibrary Display
 """
 
+import os
+import typing
+from pathlib import Path
+from PySide2.QtCore import QModelIndex, QTimeZone, Qt, QSize
+from PySide2.QtGui import QIcon, QPixmap
 from PySide2.QtWidgets import QFileSystemModel
+from qiskit_metal._gui.utility.utils import findProperty
 
 
 class QFileSystemLibraryModel(QFileSystemModel):
     """
     File System Model for displaying QLibrary in MetalGUI
 
     """
-    FILENAME = 0  # Column index to display filenames
-    pass
+    path_imgs = None  # type: Path
+    FILENAME = 0
+    imageCache = dict()
+    nameCache = dict()
+    defaultFilename = None
+    size = 64
+
+    def __init__(self, path_imgs):
+        super().__init__()
+        self.path_imgs = path_imgs
+        self.defaultFilename = str(path_imgs / "metal_logo.png")
+        pixmap = QPixmap(self.defaultFilename).scaled(
+            QSize(self.size, self.size), Qt.KeepAspectRatio,
+            Qt.SmoothTransformation)
+        self.imageCache[self.defaultFilename] = pixmap
+
+    def data(self,
+             index: QModelIndex,
+             role: int = Qt.DisplayRole) -> typing.Any:
+        """
+        Sets standard size hint for indexes and allows
+        Args:
+            index(QModelIndex): Model Index holding data
+            role(int): DisplayRole being requested of index
+        Returns:
+            Any (QVariant): Data stored under the given role for the item referred to by the index.
+        """
+
+        # There are multiple columns so we don't want display an
+        # image in each one, (see filterAcceptsColumn)
+        if index.column() == 0:
+            if not self.isDir(index):
+                qfileinfo = self.fileInfo(index)
+                absoluteFilename = str(qfileinfo.absoluteFilePath())
+                if role == Qt.DecorationRole:
+                    matches = findProperty(absoluteFilename,
```

Comment:
Oh i see, you used the docstring image. Hmm, i guess this could be more elegant!

I would just make sure the regex works for all the cases we have, and an extra space or tab wont mess with it